src/cmds/scripts/win_postinstall.py

```diff
@@ -0,0 +1,632 @@
+                                     "exec",
+                                     "pgsql",
+                                     "bin",
+                                     "createdb.exe") \
+                      + "\" -p 15007 pbs_datastore\r\n"
+        crtdb.write(buffer)
+        crtdb.write(ErrChk)
+        # install schema
+        buffer = "\"" + os.path.join(
+            pbs_path,
+            "exec",
+            "pgsql",
+            "bin",
+            "psql.exe") + "\" -p 15007 -d pbs_datastore -f " \
+                        + "\"" + os.path.join(pbs_path,
+                                              "exec",
+                                              "etc",
+                                              "pbs_db_schema.sql") \
+                        + "\"\r\n"
+        crtdb.write(buffer)
+        crtdb.write(ErrChk)
+        # create superuser
+        buffer = "\"" + os.path.join(pbs_path,
+                                     "exec",
+                                     "pgsql",
+                                     "bin",
+                                     "psql.exe") \
+                      + "\" -p 15007 -d pbs_datastore -c \"create user \\\""\
+                      + username + "\\\" SUPERUSER LOGIN\"\r\n"
+        crtdb.write(buffer)
+        # run Pbs_ds_password
+        buffer = "\"" + os.path.join(pbs_path,
+                                     "exec",
+                                     "sbin",
+                                     "pbs_ds_password.exe") + "\" -r\r\n"
+        crtdb.write(buffer)
+        crtdb.write(ErrChk)
+        # stop database
+        buffer = "call \"" + \
+            os.path.join(pbs_path, "exec", "sbin",
+                         "pbs_dataservice") + "\" stop\r\n"
+        crtdb.write(buffer)
+        crtdb.write(ErrChk)
+        # error handling
+        crtdb.write("exit /b 0\r\n")
+        crtdb.write(":ERR\r\n")
+        # stop database
+        buffer = "call \"" + \
+            os.path.join(pbs_path, "exec", "sbin",
+                         "pbs_dataservice") + "\" stop\r\n"
+        crtdb.write(buffer)
+        crtdb.write("exit /b 1\r\n")
+        crtdb.close()
+        print "Executing createdbfile as install user"
+        with open(os.devnull, 'w') as fd:
+            theproc = subprocess.Popen([createdbfile], stdout=fd, stderr=fd)
+            theproc.communicate()
+            if theproc.returncode > 0:
+                print "Failed to initilize database"
+                sys.exit(6)
+            else:
+                print "Database intialization is success"
+        UpdateDSconfig()
+        print "InstallDataService completed "
+
+
+def CreateServerdbFile():
+    with open(os.devnull, 'w') as fd:
+        theproc = subprocess.Popen(
+            ["net", "stop", "pbs_server"], stdout=fd, stderr=fd)
+        theproc.communicate()
+    # Creating PBS Server database...
+    ServerPath = os.path.join(pbs_path, "exec", "sbin", "pbs_server.exe")
```

Comment:
Just suggestion: (take it or leave it, up to you)
nit: serverpath instead ServerPath